viewing_party/party.py

```diff
@@ -1,23 +1,146 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating:
+        return None
+    new_movie = {"title": title, "genre": genre, "rating": rating}
+    return new_movie
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, movie):
+    for movies in user_data["watchlist"]:
+        if movies["title"] == movie:
+            user_data["watched"].append(movies)
+            user_data["watchlist"].remove(movies)
+    return user_data
+
+
+user_data = {"watched": [], "watchlist": [{
+    "title": "MOVIE_TITLE_1",
+    "genre": "GENRE_1",
+    "rating": "RATING_1"
+}]}
+movie = {
+    "title": "MOVIE_TITLE_1",
+    "genre": "GENRE_1",
+    "rating": "RATING_1"
+}
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    total = 0
+    count = 0
+    if not user_data["watched"]:
+        return 0
+    for data in user_data["watched"]:
+        total += data["rating"]
+        count += 1
+    average = total / count
+
+    return average
+
+
+def get_most_watched_genre(user_data):
+    if not user_data["watched"]:
+        return None
+    
+    genre_counts = {}  
+    top_genre = None   
+    top_count = 0     
+
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+
+        if genre in genre_counts:  # can maybe do this instead: genre_counts[genre] = genre_counts.get(genre, 0) + 1
+            genre_counts[genre] += 1
+        else:
+            genre_counts[genre] = 1

```

Comment:
👍 Nice use of a frequency map to count up the appearances of each genre in a single pass through the list of movies. This has better time complexity than trying to count the occurrences of each genre while looking for the most frequent genre, which results in more nested iterations. Instead, we can trade some space to store the results of counting them all once.